app/src/main/java/com/example/home/stream/StreamFragment.java

```diff
@@ -0,0 +1,133 @@
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Headers;
+
+
+public class StreamFragment extends Fragment {
+
+    public String mNoOfBedrooms;
+    public String mPropertyTypeText;
+    public Integer mZipCode;
+    public List<Home> mHome;
+    public HomeAdapter adapter;
+    public RecyclerView streamRecyclerView;
+    public LinearLayout StreamLinearLayout;
+    public static final String LATITUDE_PARAM = "latitude";
+    public static final String LONGITUDE_PARAM = "longitude";
+    public static final String RADIUS_PARAM = "radius";
+    public static final String MIN_BEDS_PARAM = "minbeds";
+    public static final String MAX_BEDS_PARAM = "maxbeds";
+    public static final String PROPERTY_TYPE_PARAM = "propertytype";
+
+
+    public StreamFragment(String mNoOfBedrooms, String mPropertyTypeText, Integer mZipCode) {
+        this.mNoOfBedrooms = mNoOfBedrooms;
+        this.mPropertyTypeText = mPropertyTypeText;
+        this.mZipCode = mZipCode;
+        // Required empty public constructor
+    }
+
+    // TODO: Rename and change types and number of parameters
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_stream, container, false);
+
+        return view;
+
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        streamRecyclerView = view.findViewById(R.id.HomesRecyclerView);
+        StreamLinearLayout = view.findViewById(R.id.StreamLinearLayout);
+        //initializes list of homes and adapter
+        mHome = new ArrayList<>();
+        //recycler view setup
+        adapter = new HomeAdapter(getContext(), mHome);
+        streamRecyclerView.setAdapter(adapter);
+        streamRecyclerView.setLayoutManager(new LinearLayoutManager(StreamLinearLayout.getContext()));
+        AsyncHttpClient client = new AsyncHttpClient();
+        RequestHeaders headers = new RequestHeaders();
+        headers.put("apikey", getContext().getString(R.string.attomdata_api_key));
+        RequestParams params = new RequestParams();
+
+        params.put(LATITUDE_PARAM, "47.628891");
+        params.put(LONGITUDE_PARAM, "-122.341408");
+        params.put(RADIUS_PARAM, "1");
+        params.put(MIN_BEDS_PARAM, mNoOfBedrooms);
+        params.put(MAX_BEDS_PARAM, mNoOfBedrooms);
+        params.put(PROPERTY_TYPE_PARAM, mPropertyTypeText);
+        client.get("https://api.gateway.attomdata.com/propertyapi/v1.0.0/property/snapshot?", headers, params, new JsonHttpResponseHandler() {
```

Comment:
I would suggest creating a Class that specifically handles this logic. What you can do is receive a User Preference object, and this class makes sure to put all the corresponding params and headers in the request. 

I believe you have a couple exmaples in the twitter app.

Reply:
Let's also think about how we can add a loading indicator

Reply:
I was talking about this with someone over breakfast today.